src/components/sbb-toggle-check/sbb-toggle-check.tsx

```diff
@@ -78,19 +79,18 @@ export class SbbToggleCheck implements AccessibilityProperties {
           aria-describedby={this.accessibilityDescribedby}
           aria-labelledby={this.accessibilityLabelledby}
         />
-        {this.labelPosition === 'before' ? <slot /> : ''}
-        <span class={`toggle-check__slider toggle-check__slider--${this.labelPosition}`}>
-          <span class="toggle-check__circle">
-            {this.checked ? (
-              <slot name="icon">
-                <sbb-icon name={this.icon}></sbb-icon>
-              </slot>
-            ) : (
-              ''
-            )}
+        <span class={'toggle-check__container'}>
+          <slot />
+          <span class={`toggle-check__slider`}>
+            <span class="toggle-check__circle">
+              {this.checked && (
```

Comment:
This is probably a nitpick; I've noticed that due to this being loaded on demand, the tick is only visible after a few milliseconds after setting it to checked. Could we maybe use display: none to hide the tick instead of removing it from the DOM entirely?